federated_content_connector/course_metadata_importer.py

```diff
@@ -222,30 +215,37 @@ def process_courses_details(cls, courses_details, courserun_with_course_uuids):
 
             enroll_by = start_date = end_date = None
 
-            if course_type in (EXEC_ED_COURSE_TYPE, BOOTCAMP_2U):
-                additional_metadata = course_metadata.get('additional_metadata')
-                if additional_metadata:
-                    enroll_by = additional_metadata.get('registration_deadline')
-                    start_date = additional_metadata.get('start_date')
-                    end_date = additional_metadata.get('end_date')
+            course_run = cls.find_attr(course_metadata.get('course_runs'), 'key', courserun_key)
+            if not course_run:
+                logger.info(
+                    f'[{cls.LOG_PREFIX}] Courserun not found. CourserunKey: {courserun_key}, CourseUUID: {course_uuid}'
+                )
+                continue
```

Comment:
Should we move this block of code up a little bit, to be directly after the part where we check for the existence of `course_metadata`? We don't need a logical branch based on `course_type` anymore.

Reply:
Sounds good; I've moved the lines where we create the `course_type` and `product_source` variables further down, right above where we pass them into `course_data`. I don't think we can remove the variables altogether without more refactoring though given they are returned are part of the `courses` dictionary.